src/main/java/mate/jdbc/dao/ManufacturerDaoImpl.java

```diff
@@ -0,0 +1,111 @@
+package mate.jdbc.dao;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import mate.jdbc.lib.Dao;
+import mate.jdbc.model.Manufacturer;
+import mate.jdbc.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoImpl implements ManufacturerDao {
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement createStatement =
+                         connection.prepareStatement(
+                             "INSERT INTO manufacturers(name, country) values(?,?);",
+                             Statement.RETURN_GENERATED_KEYS)) {
+            createStatement.setString(1, manufacturer.getName());
+            createStatement.setString(2, manufacturer.getCountry());
+            ResultSet generatedKeys = createStatement.getGeneratedKeys();
+            if (generatedKeys.next()) {
+                Long id = generatedKeys.getLong(1);
+                manufacturer.setId(id);
+            }
+            createStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new RuntimeException("Error");
+        }
+        return manufacturer;
+    }
+
+    @Override
+    public Manufacturer get(Long id) {
+        Manufacturer manufacturer = new Manufacturer();
+        String getManufacturerRequest = "SELECT * FROM manufacturers WHERE id = ?";
+        try (Connection connection = ConnectionUtil.getConnection();
+                 PreparedStatement getStatement =
+                         connection.prepareStatement(getManufacturerRequest)) {
+            getStatement.setLong(1, id);
+            ResultSet resultSet = getStatement.executeQuery();
+            if (resultSet.next()) {
+                Long idOfManufacturer = resultSet.getLong("id");
```

Comment:
It is better to be consistent in naming, if we do not use "ofManufacturer" in the remaining variables, it is not needed for id either.

To properly handle null use:

```suggestion
                Long id = resultSet.getObject("id", Long.class);
```

see: https://mate-academy.github.io/jv-program-common-mistakes/java-JDBC/jdbc-intro/JDBC-intro_checklist.html